src/action-creators/newThought.ts

```diff
@@ -37,11 +36,15 @@ const newThought = ({ offset, preventSplit, value = '' }: { offset: number, prev
 
   const showContexts = cursor && isContextViewActive(state, contextOf(pathToContext(cursor)))
 
+  const context = cursor && (showContexts && cursor.length > 2 ? pathToContext(contextOf(contextOf(cursor)))
+    : !showContexts && cursor.length > 1 ? pathToContext(contextOf(cursor))
+    : [ROOT_TOKEN])
+  const siblings = context && getThoughts(state, context)
+  const isDuplicateOnSplit = siblings && editingValue && siblings.some(sibling => sibling.value === editingValue.substring(0, offset) || sibling.value === editingValue.substring(offset))
```

Comment:
Let's make `isDuplicateOnSplit` a function (in the same scope, not in a separate file). That will provide better encapsulation of its inputs and a helpful JSDOC.